libvast/vast/flatbuffer.hpp

```diff
@@ -0,0 +1,389 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "vast/fwd.hpp"
+
+#include "vast/chunk.hpp"
+
+#include <flatbuffers/flatbuffers.h>
+#include <fmt/core.h>
+
+#include <compare>
+
+namespace vast {
+
+/// Determines whether the FlatBuffers table is a root or a child table, i.e.,
+/// whether the only data contained in the owned chunk is the table itself
+/// (root) or other the table is just part of a bigger root table (child).
+enum class flatbuffer_type {
+  root,  ///< The table type is a root type, or a nested FlatBuffers table.
+  child, ///< The table is sliced from a root table.
+};
+
+/// Determines whether the FlatBuffers table has buffer- or pointer-like
+/// behavior, i.e., whether comparison operators compare the buffer or the table
+/// pointer.
+enum class flatbuffer_semantics {
+  buffer,  ///< Comparison operators compare the buffer contents bytewise. This
+           ///< is only supported for root tables.
+  pointer, ///< Comparison operators compare the table pointers.
+};
+
+/// A wrapper class around a FlatBuffers table that allows for sharing the
+/// lifetime with the chunk containing the table.
+/// @tparam Table The generated FlatBuffers table type.
+/// @tparam Semantics Determins whether the table behaves like a buffer or a
```

Comment:
We want to get rid of this and also remove the comparison operators entirely, because we were unsure which is right, so ultimately this should end up being decided on a case-by-case basis in the classes making use of `flatbuffer<T>`.